pageserver/src/walredo.rs

```diff
@@ -296,6 +297,99 @@ impl PostgresRedoManager {
         }
     }
 
+    async fn do_with_walredo_process<
+        F: FnOnce(Arc<Process>) -> Fut,
+        Fut: Future<Output = Result<O, Error>>,
+        O,
+    >(
+        &self,
+        pg_version: u32,
+        closure: F,
+    ) -> Result<O, Error> {
+        let proc: Arc<Process> = match self.redo_process.get_or_init_detached().await {
+            Ok(guard) => match &*guard {
+                ProcessOnceCell::Spawned(proc) => Arc::clone(proc),
+                ProcessOnceCell::ManagerShutDown => {
+                    return Err(Error::Cancelled);
+                }
+            },
+            Err(permit) => {
+                let start = Instant::now();
+                // acquire guard before spawning process, so that we don't spawn new processes
+                // if the gate is already closed.
+                let _launched_processes_guard = match self.launched_processes.enter() {
+                            Ok(guard) => guard,
+                            Err(GateError::GateClosed) => unreachable!(
+                                "shutdown sets the once cell to `ManagerShutDown` state before closing the gate"
+                            ),
+                        };
```

Comment:
it seems rustfmt has already given up. could you remove the two indentation levels?

Reply:
This could also be:

```
let Ok(_permit) = self.launched_processes.enter() else { unreachable!(...) };
```